### DISCUSSION TITLE
Get VLANs on a device

### DISCUSSION THREAD

**Discussion by misch42** [netbox-community/netbox#14007] [General]
I want to assign vlans to devices via jinja config templates.

The function device.site.vlans.all() work and gives me a VLANs for a specific device if VLANs are assigned to a site where the device is located.

BUT: we also have VLANs organized in vlan groups. These groups are assigned to site groups. So device.site.vlans.all() will not find it. any ideas how to solve the problem?

**Comment by misch42** [comment]
I found the API endpoint ipam.vlans/?available_on_device=

But how can I use this in a jinja config template?

**Comment by T-Hollenbeck** [comment]
Hi, ich have the same problem, that we have organized the vlan in groups.

Alternative it is possible to become alle vlans from a group that a linked to a site or sitegroup and get this via a variable like device.site.vlans.all()

**Reply by T-Hollenbeck** [reply]
For me I have find a solution, 
here ist the code to filter by VLAN by group

```
{%- for vlan in ipam.VLAN.objects.all() | selectattr('group.name', 'equalto', 'GROUP') %}
VLAN ID: {{ vlan.vid }} - Name: {{ vlan.name }} - {{vlan.group.name}} - Scope: {{ vlan.group.scope }}
{% endfor %}
```

**Comment by wz4** [comment]
You can retrieve all VLANs available on a device using:

```
{% for v in ipam.VLAN.objects.all().get_for_device(device) -%}
    {{ v.vid }}
{% endfor %}
```

**Reply by qaxi** [reply]
Are those "magic" documented somewhere?
I fight with that too.

**Comment by dejantep** [comment]
Is this applicable to already assigned VLANs to device? if i want to get all assigned VLANs per specific device
